pkg/apis/gcp/validation/worker.go

```diff
@@ -120,22 +125,54 @@ func validateDataVolume(workerConfig *gcp.WorkerConfig, volume core.DataVolume,
 		allErrs = append(allErrs, field.Required(fldPath.Child("type"), "must not be empty"))
 		return allErrs
 	}
-	if *volume.Type == worker.VolumeTypeScratch {
+
+	allErrs = append(allErrs, validateScratchDisk(*volume.Type, workerConfig)...)
+
+	return allErrs
+}
+
+func validateScratchDisk(volumeType string, workerConfig *gcp.WorkerConfig) field.ErrorList {
```

Comment:
(apologies for delay in reviewing. was on leave.

Just to confirm :I assume this new validation routine was introduced since we didn't have one before to enforce this section in the usage docuement:

"If you attach the disk with SCRATCH type, either an NVMe interface or a SCSI interface must be specified. It is only meaningful to provide this volume interface if only SCRATCH data volumes are used."

Reply:
Previously, the validation for the SCRATCH disk type was present and has now been slightly refined. 
A new validation rule requires that the workerConfigDataVolume name must correspond to a dataVolume name.